.github/workflows/test-go.yml

```diff
@@ -0,0 +1,51 @@
+name: test-go
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  workflow_call:
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        go-version: [ 1.17.x ]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Go cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+          key: ${{ runner.os }}-go-cache-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-cache
+      - name: Run gofmt
+        working-directory: go
+        run: gofmt -d -e . 2>&1 | tee outfile && test -z "$(cat outfile)" && rm outfile
+      # - name: Run staticcheck
+      #   if: matrix.go-version == '1.17.x'
+      #   run: |
+      #     bin/staticcheck_linux_amd64_2021.1.2 github.com/cucumber/ci-environment/go
+      - name: Run go vet
+        working-directory: go
+        run: go vet ./...
+      - name: Run go test
+        working-directory: go
+        run: go test ./...
```

Comment:
Maybe we could use `make test` to make sure to use the latest `ciEnvironment.json`?

Reply:
Thanks for the quick feedback. I can get those updated soon.  For the acceptance tests, would it be helpful to rename and/or restructure the TestDetectCIEnvironment function in detect_environment_test.go, or is that not what we're looking for with regards to acceptance tests?  In the TestDetectCIEnvironment  function, the loadTestData() function reads the .txt and .json files in the ../testdata directory and creates a slice of inputs and expected outputs that are iterated over in a series of subtests, 1 per environment.  Those examples were actually extremely useful in pointing out some of the things that I initially didn't handle, like wildcards in some of the environment variable names.

Reply:
Ok, I have some updates per the review. If there are other changes required, just let me know, including if it would be more clear to chose a different naming convention for the acceptance tests. One item with those that I'd like to validate is that the expression for git.revision for GitHub Actions in CiEnvironments.json is ${GITHUB_SHA}. In GitHubActionsPullRequestOpened.txt and GitHubActionsPullRequestSynchronize.txt, they both have GITHUB_SHA=99684bcacf01d95875834d87903dcb072306c9ad. But in the accompanying .txt.json files, the git.revision is 99684bcacf01d95875834d87903dcb072306c9ad and 3738117e3337e54955580f4e98cf767d96b42135, respectively. But in the GitHubActions.txt/GitHubActions.txt.json, the GITHUB_SHA and git.revision match.  I may be missing something in my implementation. Why does the git.revision not match the GITHUB_SHA in those two cases?

Reply:
@dumpsterfireproject you might find the explanation here: https://github.com/cucumber/ci-environment/issues/67